herddb-core/src/main/java/herddb/client/HDBClient.java

```diff
@@ -154,10 +154,16 @@ void releaseConnection(HDBConnection connection) {
     }
 
     Channel createChannelTo(ServerHostData server, ChannelEventListener eventReceiver) throws IOException {
-        int timeoutms = configuration.getInt(ClientConfiguration.PROPERTY_NETWORK_TIMEOUT, ClientConfiguration.PROPERTY_NETWORK_TIMEOUT_DEFAULT);
-        int timeouts = (int) TimeUnit.MILLISECONDS.toSeconds(timeoutms);
-        return NettyConnector.connect(server.getHost(), server.getPort(), server.isSsl(), timeoutms, timeouts, eventReceiver, thredpool,
-                networkGroup);
+        final int timeoutms = configuration.getInt(ClientConfiguration.PROPERTY_NETWORK_TIMEOUT, ClientConfiguration.PROPERTY_NETWORK_TIMEOUT_DEFAULT);
+        final int timeouts = (int) TimeUnit.MILLISECONDS.toSeconds(timeoutms);
+        final boolean connectLocalVmAllowed = configuration.getBoolean(ClientConfiguration.PROPERTY_CLIENT_CONNECT_LOCALVM_SERVER, ClientConfiguration.PROPERTY_CLIENT_CONNECT_LOCALVM_SERVER_DEFAULT);
+        if (!connectLocalVmAllowed) {
```

Comment:
inverted if?

Reply:
no, it isn't

if you allow LocalVMConnection then we you  `NettyConnector.connect` that detects if LocalVM is available.
if you do not allow LocalVMConnection then we force a network connection `NettyConnector.connectUsingNetwork`